service/backend/pchain/block.go

```diff
@@ -299,62 +285,13 @@ func (b *Backend) getChainIDs(ctx context.Context) (map[string]string, error) {
 	return b.chainIDs, nil
 }
 
-func (b *Backend) getBlockDetails(ctx context.Context, index int64, hash string) (*indexer.ParsedBlock, error) {
-	if index <= 0 && hash == "" {
-		return nil, errMissingBlockIndexHash
-	}
-
-	blockHeight := uint64(index)
-
-	// Extract block id from hash parameter if it is non-empty, or from index if stated
-	if hash != "" {
-		height, err := b.getBlockHeight(ctx, hash)
-		if err != nil {
-			return nil, err
-		}
-
-		if blockHeight > 0 && height != blockHeight {
-			return nil, errMismatchedHeight
-		}
-		blockHeight = height
-	}
-
-	return b.indexerParser.ParseBlockAtHeight(ctx, blockHeight)
-}
-
-func (b *Backend) getBlockHeight(ctx context.Context, hash string) (uint64, error) {
-	blockID, err := ids.FromString(hash)
-	if err != nil {
-		return 0, err
-	}
-
-	blockBytes, err := b.pClient.GetBlock(ctx, blockID)
-	if err != nil {
-		return 0, err
-	}
-
-	var block blocks.Block
-	_, err = b.codec.Unmarshal(blockBytes, &block)
-	if err != nil {
-		return 0, err
-	}
-
-	return block.Height(), nil
-}
-
 func (b *Backend) isGenesisBlockRequest(ctx context.Context, index int64, hash string) (bool, error) {
 	genesisBlock, err := b.getGenesisBlock(ctx)
 	if err != nil {
 		return false, err
 	}
 
-	// if hash is provided, make sure it matches genesis block hash
-	if hash != "" {
-		return hash == genesisBlock.BlockID.String(), nil
-	}
-
-	// if hash is omitted, check if the height matches the genesis block height
-	return index == int64(genesisBlock.Height), nil
+	return hash == genesisBlock.BlockID.String() || index == int64(genesisBlock.Height), nil
```

Comment:
I still don't see how this change is not equivalent no the first one

Reply:
Requests for /block can specify either height or hash. If a request only specifies a non-genesis hash, the height will be defaulted to 0, and the index passed down to here will be 0 and will match the genesis height, even though the request is for a hash that is not for the genesis block. 

If you really want to write this in a single line, you probably should do something like:

```
return hash == genesisBlock.BlockID.String() || hash == "" && index == int64(genesisBlock.Height), nil
```

Reply:
Thanks for the patient explaination, I could not see the default case. Rolling back you initial implementation.